src/core/biblio.js

```diff
@@ -115,85 +115,89 @@ export function stringifyReference(ref) {
-    for (var j = 0; j < refs.length; j++) {
-      var ref = refs[j];
-      $("<dt></dt>")
-        .attr({ id: "bib-" + ref })
-        .text("[" + ref + "]")
-        .appendTo($dl);
-      var $dd = $("<dd></dd>").appendTo($dl);
-      var refcontent = conf.biblio[ref];
-      var circular = {};
-      var key = ref;
-      circular[ref] = true;
-      while (refcontent && refcontent.aliasOf) {
-        if (circular[refcontent.aliasOf]) {
-          refcontent = null;
-          const msg = `Circular reference in biblio DB between [\`${ref}\`] and [\`${key}\`].`;
-          pub("error", msg);
-        } else {
-          key = refcontent.aliasOf;
-          refcontent = conf.biblio[key];
-          circular[key] = true;
-        }
-      }
-      aliases[key] = aliases[key] || [];
-      if (aliases[key].indexOf(ref) < 0) aliases[key].push(ref);
-      if (refcontent) {
-        $dd.html(stringifyReference(refcontent) + "\n");
-      } else {
-        if (!badrefs[ref]) badrefs[ref] = 0;
-        badrefs[ref]++;
-        $dd.html("<em style='color: #f00'>Reference not found.</em>\n");
-      }
-    }
+    sec.appendChild(hyperHTML`
+      <dl class='bibliography'>
+        ${refs.map(addRef)}
+      </dl>`);
+
+    refsec.appendChild(sec);
   }
-  for (var k in aliases) {
+  document.body.appendChild(refsec);
+
+  for (const k in aliases) {
     if (aliases[k].length > 1) {
       let msg = `[${k}] is referenced in ${aliases[k].length} ways: `;
-      msg += `(${aliases[k].map(item => `'${item}'`).join(", ")}). This causes`;
-      msg += ` duplicate entries in the References section.`;
+      msg += `(${aliases[k].map(item => `'${item}'`).join(", ")}). `;
+      msg += `This causes duplicate entries in the References section.`;
       pub("warn", msg);
     }
   }
-  for (var item in badrefs) {
+
+  for (const item in badrefs) {
     const msg = `Bad reference: [\`${item}\`] (appears ${badrefs[item]} times)`;
     if (badrefs.hasOwnProperty(item)) pub("error", msg);
   }
+
+  function addRef(ref) {
+    let refcontent = conf.biblio[ref];
+    let key = ref;
+    const circular = {};
+    circular[ref] = true;
+    while (refcontent && refcontent.aliasOf) {
+      if (circular[refcontent.aliasOf]) {
+        refcontent = null;
+        const msg = `Circular reference in biblio DB between [\`${ref}\`] and [\`${key}\`].`;
+        pub("error", msg);
+      } else {
+        key = refcontent.aliasOf;
+        refcontent = conf.biblio[key];
+        circular[key] = true;
+      }
+    }
+    aliases[key] = aliases[key] || [];
+    if (!aliases[key].includes(ref)) aliases[key].push(ref);
+    const dtId = "bib-" + ref;
+    let output = `<dt id="${dtId}">[${ref}]</dt>`;
+    if (refcontent) {
+      output += `<dd>${stringifyReference(refcontent)}</dd>`;
+    } else {
+      if (!badrefs[ref]) badrefs[ref] = 0;
+      badrefs[ref]++;
+      output += `<dd><em style='color: #f00'>Reference not found.</em></dd>`;
```

Comment:
we should get rid of `"style='color: #f00'"` and add a CSS class (or use `respec-offending-element` or whatever it's called).